src/main/java/org/folio/circulation/support/FindWithMultipleCqlIndexValues.java

```diff
@@ -0,0 +1,18 @@
+package org.folio.circulation.support;
+
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.MultipleRecords;
+import org.folio.circulation.support.http.client.CqlQuery;
+
+public interface FindWithMultipleCqlIndexValues<T> {
+    CompletableFuture<Result<MultipleRecords<T>>> findByIds(
+      Collection<String> ids);
+
+    CompletableFuture<Result<MultipleRecords<T>>> findByIndexName(
+      Collection<String> ids, String indexName);
+
+    CompletableFuture<Result<MultipleRecords<T>>> findByIdIndexAndQuery(
+      Collection<String> ids, String indexName, Result<CqlQuery> andQuery);
```

Comment:
Good suggestion. I think that is a possibility.

Would only one of `forIds` or `forIndex` would be allowed?

My initial concern with that, is that it would require additional code paths behind the scenes. I might experiment and see if that can improve the structure of the code.

Reply:
The intent behind this change was to change as little of the external interface as possible.

This is quite a significant change to the external interface. I might experiment with implementing this as an alternative method, and delegate the others to it behind the scenes.

Then a later change can migrate all of the production code to this new way.

What do you think?

Reply:
Thanks, works for me. 

> Would only one of forIds or forIndex would be allowed? 

I think we should allow only one of them.

Reply:
I've done a small experiment with this. Please let me know your thoughts

Reply:
Thank you, looks good for me.